addons/account/models/account_move_line.py

```diff
@@ -2598,6 +2598,8 @@ def _prepare_exchange_difference_move_vals(self, amounts_list, company=None, exc
             return
 
         journal = company.currency_exchange_journal_id
+        if not journal:
+            raise UserError(_('Please set the "Exchange Gain or Loss" journal in your company settings in order to register this payment'))
```

Comment:
Hello @epictete , Thank you for the review, I've adapted the code to the requested changes.
regarding your question about targeting the oldest supported version, before Odoo 17, this bug wasn't present because a recent change caused the issue: https://github.com/odoo/odoo/blob/17.0/addons/account/models/account_move_line.py#L2613
So we cannot reproduce the bug in any version before Odoo 17

Reply:
Thanks for the update @amay-odoo, LGTM :ok_hand: 

> Could you please add the detailed steps to reproduce the issue?

Reply:
Hello @epictete , Thank you for the review, I've added how to reproduce part to the commit message, could you please look at it when you are available? Thanks